src/client/api/metrics.c

```diff
@@ -0,0 +1,914 @@
+	}
+	D_MUTEX_UNLOCK(&metrics_tls_lock);
+	if (stats->u.st_obj_update.st_min == ULONG_MAX) {
+		stats->u.st_obj_update.st_min = 0;
+	}
+	if (stats->u.st_obj_fetch.st_min == ULONG_MAX) {
+		stats->u.st_obj_fetch.st_min = 0;
+	}
+out:
+	return rc;
+}
+
+int
+daos_metrics_alloc_distbuf(daos_metrics_udists_t **dists)
+{
+	int rc = 0;
+	D_ALLOC_PTR(*dists);
+	if (*dists == NULL) {
+		D_ERROR("Failed to allocate memory\n");
+		rc = -DER_NOMEM;
+	}
+	return rc;
+}
+
+int
+daos_metrics_free_distbuf(daos_metrics_udists_t *dists)
+{
+	free(dists);
+	return 0;
+}
+
+int
+daos_metrics_get_dist(enum daos_metrics_dist_grp md_grp, daos_metrics_udists_t *dist)
+{
+	int rc = 0, i;
+	dc_metrics_tls_data_t *entry;
+
+	if (is_metrics_enabled == 0) {
+		D_GOTO(out, rc = 0);
+	}
+	memset(dist, 0, sizeof(*dist));
+	D_MUTEX_LOCK(&metrics_tls_lock);
+	switch(md_grp) {
+		case DAOS_METRICS_DIST_IO_BSZ:
+			dist->md_grp = DAOS_METRICS_DIST_IO_BSZ;
+			d_list_for_each_entry(entry, &metrics_list, list) {
+				dist->u.dt_bsz.ids_updatesz += entry->ids.ids_updatesz;
+				dist->u.dt_bsz.ids_fetchsz  += entry->ids.ids_fetchsz;
+				for (i=0;i<DAOS_METRICS_IO_BYSIZE_COUNT;i++) {
+					dist->u.dt_bsz.ids_updatecnt_bkt[i] += \
+						entry->ids.ids_updatecnt_bkt[i];
+					dist->u.dt_bsz.ids_fetchcnt_bkt[i]  += \
+						entry->ids.ids_fetchcnt_bkt[i];
+				}
+			}
+			break;
+		case DAOS_METRICS_DIST_IO_BPT:
+			dist->md_grp = DAOS_METRICS_DIST_IO_BPT;
+			d_list_for_each_entry(entry, &metrics_list, list) {
+				dist->u.dt_bpt.idp_updatesz += entry->idp.idp_updatesz;
+				for (i=0;i<DAOS_METRICS_IO_BYPTYPE_COUNT;i++) {
+					dist->u.dt_bpt.idp_updatecnt_bkt[i] += \
+						entry->idp.idp_updatecnt_bkt[i];
+					dist->u.dt_bpt.idp_updatesz_bkt[i]  += \
+						entry->idp.idp_updatesz_bkt[i];
+				}
+			}
+			break;
+		default:
+			D_ERROR("Invalid argument md_grp = %d\n", md_grp);
+			rc = -DER_INVAL;
+	}
+	D_MUTEX_UNLOCK(&metrics_tls_lock);
+out:
+	return rc;
+}
+
+static int
+dump_pool_rpccntrs(FILE *fp)
+{
+	int rc;
+	daos_metrics_ucntrs_t *cntrs = NULL;
+	daos_metrics_pool_rpc_cntrs_t *pcntrs;
+
+	rc = daos_metrics_alloc_cntrsbuf(&cntrs);
+	if (rc != 0) {
+		D_ERROR("Failed to dump pool rpc counters rc = %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	rc = daos_metrics_get_cntrs(DAOS_METRICS_POOL_RPC_CNTR, cntrs);
+	if (rc != 0) {
+		D_ERROR("Failed to dump pool rpc counters rc = %d\n", rc);
+		D_GOTO(alloc_out, rc);
+	}
+
+	pcntrs = &cntrs->u.arc_pool_cntrs;
+
+	fprintf(fp, "********************  Dumping Pool RPC Counters *********************\n");
+	fprintf(fp, "%-16s\t%12s\t%12s\t%12s\n","Name","Inflight","Success","Failure");
+	fprintf(fp, "%-16s\t%12lu\t%12lu\t%12lu\n", "pool connect", pcntrs->prc_connect_cnt.mc_inflight, \
+			pcntrs->prc_connect_cnt.mc_success, pcntrs->prc_connect_cnt.mc_failure);
+	fprintf(fp, "%-16s\t%12lu\t%12lu\t%12lu\n","pool disconnect", pcntrs->prc_disconnect_cnt.mc_inflight,\
+			pcntrs->prc_disconnect_cnt.mc_success,pcntrs->prc_disconnect_cnt.mc_failure);
```

Comment:
(style)  line over 100 characters
(style)  space required after that ',' (ctx:VxV)